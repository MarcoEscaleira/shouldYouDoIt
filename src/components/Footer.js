import React from 'react';

const Footer = () => (
  <footer className="footer">
    <p className="footer__message">
      Made with <span className="message--red">&#x2665;</span> by <a href="https://github.com/marcoescaleira" className="message__name">Marco Escaleira</a> & <a href="https://github.com/MarceloSilva00" className="message__name">Marcelo Silva</a> 
      </p>
  </footer>
);

export default Footer;