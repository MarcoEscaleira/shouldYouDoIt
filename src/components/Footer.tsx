import React from 'react';

const Footer: React.FC = () => (
  <footer className="footer">
    <p className="footer__message">
      Made with <span className="message--red">&#x2665;</span> by&nbsp;
      <a href="https://github.com/marcoescaleira" className="message__name">
        Marco Escaleira
      </a>&nbsp;
      &&nbsp;
      <a href="https://github.com/MarceloSilva00" className="message__name">
        Marcelo Silva
      </a>
    </p>
  </footer>
);

export default Footer;
