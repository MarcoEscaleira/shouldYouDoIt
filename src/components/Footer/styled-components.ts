import styled from "styled-components";
import {
  sizes,
  pallete,
  footerHeight,
  footerDesktopHeight
} from "~/styles/_settings";

export const Container = styled.footer`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: ${footerHeight};

  @media (min-width: ${sizes.desktop_breakpoint}) {
    height: ${footerDesktopHeight};
  }
`;

export const Text = styled.p`
  text-align: center;
  color: ${pallete.second_light};
  align-items: center;
  font-size: 1.3rem;

  @media (min-width: ${sizes.desktop_breakpoint}) {
    font-size: ${sizes.m};
  }
`;

export const NameLink = styled.a`
  text-decoration: underline;
  letter-spacing: 0.15rem;
  color: ${pallete.second_light};
  font-weight: 500;
`;

export const Heart = styled.span`
  color: red;
`;
