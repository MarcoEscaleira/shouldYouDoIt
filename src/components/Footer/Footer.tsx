import React from "react";
import { Container, Text, NameLink, Heart } from "./styled-components";

const Footer: React.FC = () => (
  <Container>
    <Text>
      Made with <Heart>&#x2665;</Heart> by&nbsp;
      <NameLink href="https://github.com/marcoescaleira">
        Marco Escaleira
      </NameLink>
      &nbsp;&&nbsp;
      <NameLink href="https://github.com/MarceloSilva00">
        Marcelo Silva
      </NameLink>
    </Text>
  </Container>
);

export default Footer;
