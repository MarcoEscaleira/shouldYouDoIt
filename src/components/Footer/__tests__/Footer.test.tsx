import React from "react";
import { render } from "@testing-library/react";
import Footer from "../Footer";

describe("Footer test suite", () => {
  it("Should render Footer correctly", () => {
    const { container } = render(<Footer />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
