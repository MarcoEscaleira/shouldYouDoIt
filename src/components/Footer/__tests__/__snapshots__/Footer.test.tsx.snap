// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer test suite Should render Footer correctly 1`] = `
<footer
  class="sc-bdVaJa bulNMW"
>
  <p
    class="sc-bwzfXH kgMNSL"
  >
    Made with 
    <span
      class="sc-bxivhb gNxrLP"
    >
      ♥
    </span>
     by 
    <a
      class="sc-htpNat kiXTiO"
      href="https://github.com/marcoescaleira"
    >
      Marco Escaleira
    </a>
     & 
    <a
      class="sc-htpNat kiXTiO"
      href="https://github.com/MarceloSilva00"
    >
      Marcelo Silva
    </a>
  </p>
</footer>
`;
